src/net/unix/network.c

```diff
@@ -79,8 +85,9 @@ char *_zn_select_scout_iface()
             current = current->ifa_next;
         } while ((iface == 0) && (current != 0));
     }
-    char *result = strdup((iface != 0) ? iface : loopback);
-    freeifaddrs(ifap);
```

Comment:
Is there any particular reason on removing 'freeifaddrs(ifap)'? 
This could lead to a memory leak where the memory allocated for 'ifap' is never freed.

Reply:
Fixed.
I forgot to put it back after debugging the zenoh-pico library for ESP32.
In any case, I still plan to go through that function to see how to support the scouting in ESP32.